corehq/motech/requests.py

```diff
@@ -43,12 +43,21 @@ def request_wrapper(self, *args, **kwargs):
 
 
 class Requests(object):
+    """
+    Wraps the requests library to simplify use with JSON REST APIs.
+
+    Sets auth headers automatically, and requests JSON responses by
+    default. To accommodate authenticating non-API requests easily, it
+    uses a single session across all requests made with the same
+    instance.
+    """
     def __init__(self, domain_name, base_url, username, password, verify=True):
         self.domain_name = domain_name
         self.base_url = base_url
         self.username = username
         self.password = password
         self.verify = verify
+        self.session = requests.Session()
```

Comment:
The previous code in this PR did some lifecycle management on this session. I'm not sure if it matters that the session never gets destroyed, but if this `Requests` object were to be re-used after the `authenticate_session` from OpenMRS is called, I think the server might believe its the same session, username and password

Reply:
You're right. I think it's OK though, because (so far) it is always the same user. If that ever changes, or we need to use different sessions, I was hoping that my docstring, "... it uses a single session across all requests made with the same instance", would help the dev. 

I really liked your idea of having `session` as a context manager method. If we ever need it, I think that's the approach I would take.